assets/blocks/course-outline/edit.js

```diff
@@ -1,78 +1,44 @@
 import { InnerBlocks } from '@wordpress/block-editor';
+import { compose } from '@wordpress/compose';
+import { useSelect, withSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { CourseOutlinePlaceholder } from './placeholder';
 
-import useBlocksCreator from './use-block-creator';
-
-// TODO: Fetch from API.
-const data = [
-	{
-		id: 1,
-		type: 'module',
-		title: 'Module 1',
-		description: 'Module description 1',
-		lessons: [
-			{
-				id: 2,
-				type: 'lesson',
-				title: 'Lesson 2',
-			},
-			{
-				id: 3,
-				type: 'lesson',
-				title: 'Lesson 3',
-			},
-		],
-	},
-	{
-		id: 9,
-		type: 'lesson',
-		title: 'Lesson 9',
-	},
-	{
-		id: 10,
-		type: 'lesson',
-		title: 'Lesson 10',
-	},
-	{
-		id: 4,
-		type: 'module',
-		title: 'Module 4',
-		description: 'Module description 4',
-		lessons: [
-			{
-				id: 5,
-				type: 'lesson',
-				title: 'Lesson 5',
-			},
-		],
-	},
-	{
-		id: 6,
-		type: 'module',
-		title: 'Module 6',
-		description: 'Module description 6',
-		lessons: [],
-	},
-	{
-		id: 7,
-		type: 'lesson',
-		title: 'Lesson 7',
-	},
-];
+import { useBlocksCreator } from './use-block-creator';
 
 /**
  * Edit course outline block component.
  *
- * @param {Object} props           Component props.
- * @param {string} props.clientId  Block client ID.
- * @param {string} props.className Custom class name.
+ * @param {Object}   props           Component props.
+ * @param {string}   props.clientId  Block client ID.
+ * @param {string}   props.className Custom class name.
+ * @param {Object[]} props.blocks    Course module and lesson blocks
  */
-const EditCourseOutlineBlock = ( { clientId, className } ) => {
-	useBlocksCreator( data, clientId );
+const EditCourseOutlineBlock = ( { clientId, className, blocks } ) => {
+	const { setBlocks } = useBlocksCreator( clientId );
+
+	useEffect( () => {
```

Comment:
Moved this effect outside the custom hook so that it can expose a `setBlocks`, which the placeholder's create course/lesson uses.